applications/RANSApplication/custom_elements/convection_diffusion_reaction_cross_wind_stabilized_element.h

```diff
@@ -0,0 +1,989 @@
+     */
+    array_1d<double, 3> EvaluateInPoint(
+        const Variable<array_1d<double, 3>>& rVariable,
+        const Vector& rShapeFunction,
+        const int Step = 0) const
+    {
+        return RansCalculationUtilities::EvaluateInPoint(
+            this->GetGeometry(), rVariable, rShapeFunction, Step);
+    }
+
+    double GetScalarVariableGradientNorm(
+        const Matrix& rShapeFunctionDerivatives,
+        const int Step = 0) const
+    {
+        KRATOS_TRY;
+
+        array_1d<double, 3> scalar_variable_gradient;
+        this->CalculateGradient(scalar_variable_gradient,
+                                TConvectionDiffusionReactionData::GetScalarVariable(),
+                                rShapeFunctionDerivatives, Step);
+        return norm_2(scalar_variable_gradient);
+
+        KRATOS_CATCH("");
+    }
+
+    double GetScalarVariableRelaxedAcceleration(
+        const Vector& rShapeFunctions,
+        const int Step = 0) const
+    {
+        KRATOS_TRY;
+
+        return this->EvaluateInPoint(
+            TConvectionDiffusionReactionData::GetScalarRelaxedRateVariable(),
+            rShapeFunctions, Step);
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief Get the Geometry Parameter Derivatives object
+     *
+     * This method calculates partial derivatives of parametric coordinates(i.e. $\underline{\xi}$) of element
+     * w.r.t. physical coordinates (i.e. $\underline{x}$)
+     *
+     * \[
+     *      \frac{\partial \underline{\xi}}{\partial \underline{x}}
+     * \]
+     *
+     * @return ShapeFunctionDerivativesArrayType
+     */
+    ShapeFunctionDerivativesArrayType GetGeometryParameterDerivatives() const
+    {
+        const auto& r_geometry = this->GetGeometry();
+        return RansCalculationUtilities::CalculateGeometryParameterDerivatives(
+            r_geometry, this->GetIntegrationMethod());
+    }
+
+    void CalculatePrimalDampingMatrix(
+        BoundedMatrix<double, TNumNodes, TNumNodes>& rDampingMatrix,
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY
+
+        rDampingMatrix.clear();
+
+        // Get Shape function data
+        Vector gauss_weights;
+        Matrix shape_functions;
+        ShapeFunctionDerivativesArrayType shape_derivatives;
+        this->CalculateGeometryData(gauss_weights, shape_functions, shape_derivatives);
+        const ShapeFunctionDerivativesArrayType& r_parameter_derivatives =
+            this->GetGeometryParameterDerivatives();
+        const IndexType num_gauss_points = gauss_weights.size();
+
+        const double delta_time = this->GetDeltaTime(rCurrentProcessInfo);
+        const double bossak_alpha = rCurrentProcessInfo[BOSSAK_ALPHA];
+        const double bossak_gamma =
+            TimeDiscretization::Bossak(bossak_alpha, 0.25, 0.5).GetGamma();
+        const double dynamic_tau = rCurrentProcessInfo[DYNAMIC_TAU];
+        const double eps = std::numeric_limits<double>::epsilon();
```

Comment:
I'd use a larger zero tolerance value in here.

Reply:
done :)